src/protocols/device_path.rs

```diff
@@ -52,3 +52,20 @@ impl Hardware {
     pub const SUBTYPE_CONTROLLER: u8 = 0x05;
     pub const SUBTYPE_BMC: u8 = 0x06;
 }
+
+#[repr(C, packed)]
```

Comment:
So far, I didn't use `packed` because it is still not fully supported in Rust. You can still get undefined behavior when referencing unaligned member fields of packed structures. For instance, in `device_path::Protocol::length` I explicitly used `[u8; 2]` rather than `u16`. This also forces the user to think about endianness.

Having said that, I do see the value in `packed` and it does make live so much easier. It is also what tianocore does, so I think I am fine with that.

Reply:
We probably want to add `#[derive(Clone, Copy, Debug)]`, though.

Reply:
I added `#[derive(Clone, Copy, Debug)]`.